File: src/ui/colors.ts
/**
 * ANSI color utilities for terminal output
 */

let colorEnabled = true;

export function setColorEnabled(enabled: boolean): void {
	colorEnabled = enabled;
}

export function isColorEnabled(): boolean {
	return colorEnabled;
}

function wrap(code: string, text: string): string {
	if (!colorEnabled) return text;
	return `\x1b[${code}m${text}\x1b[0m`;
}

// Basic colors
export const red = (text: string): string => wrap("31", text);
export const green = (text: string): string => wrap("32", text);
export const yellow = (text: string): string => wrap("33", text);
export const cyan = (text: string): string => wrap("36", text);
export const gray = (text: string): string => wrap("90", text);

// Text styles
export const bold = (text: string): string => wrap("1", text);
export const dim = (text: string): string => wrap("2", text);

File: internal/ui/colors.go
package ui

import (
	"fmt"
	"os"

	"golang.org/x/term"

	"github.com/CircleCI-Public/chunk-cli/internal/config"
)

var colorEnabled = detectColor()

func detectColor() bool {
	if os.Getenv(config.EnvNoColor) != "" {
		return false
	}
	return term.IsTerminal(int(os.Stderr.Fd()))
}

// SetColorEnabled overrides automatic color detection.
func SetColorEnabled(enabled bool) {
	colorEnabled = enabled
}

func wrap(code, text string) string {
	if !colorEnabled {
		return text
	}
	return fmt.Sprintf("\x1b[%sm%s\x1b[0m", code, text)
}

func Red(text string) string    { return wrap("31", text) }
func Green(text string) string  { return wrap("32", text) }
func Yellow(text string) string { return wrap("33", text) }
func Cyan(text string) string   { return wrap("36", text) }
func Gray(text string) string   { return wrap("90", text) }
func Bold(text string) string   { return wrap("1", text) }
func Dim(text string) string    { return wrap("2", text) }
